lib/3scale_toolbox/commands/activedocs_command/apply_command.rb

```diff
@@ -0,0 +1,130 @@
+            end
+          end
+
+          def run
+            res = activedocs
+            validate_option_params(res)
+            if !res
+              res = Entities::ActiveDocs::create(remote: remote, attrs: create_activedocs_attrs)
+            else
+              res.update(activedocs_attrs) unless activedocs_attrs.empty?
+            end
+
+            output_msg_array = ["Applied ActiveDocs id: #{res.id}"]
+            output_msg_array << "Published" if option_publish
+            output_msg_array << "Hidden" if option_hide
+            puts output_msg_array.join(";")
+          end
+
+          private
+
+          def remote
+            @remote ||= threescale_client(arguments[:remote])
+          end
+
+          def validate_option_params(adocs)
+            if !adocs
+              check_openapi_spec_defined
+            end
+
+            if option_publish && option_hide
+              raise ThreeScaleToolbox::Error.new("--publish and --hide are mutually exclusive")
+            end
+          end
+
+          def check_openapi_spec_defined
+            if !option_openapi_spec
+              raise ThreeScaleToolbox::Error.new("--openapi-spec is mandatory when ActiveDocs is created")
+            end
+          end
+
+          def activedocs
+            @activedocs ||= find_activedocs
+          end
+
+          def ref
+            arguments[:activedocs_id_or_system_name]
+          end
+
+          def find_activedocs
+            Entities::ActiveDocs.find(remote: remote, ref: ref)
+          end
+
+          def option_publish
+            !options[:publish].nil?
+          end
+
+          def option_hide
+            !options[:hide].nil?
+          end
+
+          def option_openapi_spec
+            options[:'openapi-spec']
+          end
+
+          def activedocs_json_spec
+            @json_spec ||= read_activedocs_json_spec
+          end
+
+          def read_activedocs_json_spec
+            activedoc_spec = option_openapi_spec
+            activedoc_spec_content = load_resource(activedoc_spec)
+            JSON.pretty_generate(activedoc_spec_content)
+          end
+
+          def activedocs_attrs
+            activedocs_basic_attrs.tap do |params|
+              params["body"] = activedocs_json_spec if !option_openapi_spec.nil?
+              params["published"] = true if option_publish
+              params["published"] = false if option_hide
+            end
+          end
+
+          def activedocs_basic_attrs
+            {
+              "service_id" => options[:'service-id'],
+              "skip_swagger_validations" => options[:'skip-swagger-validations'],
+              "description" => options[:'description'],
+              "system_name" => options[:'system-name'],
+              "name" => options[:name],
+            }.compact
+          end
+
+          def create_activedocs_attrs
```

Comment:
I would check if openapi is defined here `check_openapi_spec_defined`.

Or another option is do not validate it at toolbox level, and let 3scale API complain about empty required parameter.

Reply:
:+1: